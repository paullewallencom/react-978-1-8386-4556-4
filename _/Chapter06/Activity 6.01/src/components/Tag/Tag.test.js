import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Tag } from "./index";

afterEach(cleanup);

it("renders a tag", () => {
  const { asFragment } = render(
    <Tag selectedTag={null} handleTags={() => {}}>
      milk
    </Tag>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders active tag when selected tag matches current tag", () => {
  const { container } = render(
    <Tag selectedTag="milk" handleTags={() => {}}>
      milk
    </Tag>
  );
  expect(container.firstChild.classList.contains("active")).toBe(true);
});
