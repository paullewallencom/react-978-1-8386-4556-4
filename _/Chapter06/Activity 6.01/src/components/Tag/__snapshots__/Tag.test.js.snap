// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a tag 1`] = `
<DocumentFragment>
  <button
    class="tag "
  >
    milk
  </button>
</DocumentFragment>
`;
